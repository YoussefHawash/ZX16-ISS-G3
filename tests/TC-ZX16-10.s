.text
.org 0x0000

main:
        li16 a0, 254       # max volume
        ecall 5

        # — Opening Ascend: C5 → E5 → G5 → B5 → C6
        li16 a0, 523       # C5
        li16 a1, 100
        ecall 4

        li16 a0, 659       # E5
        li16 a1, 100
        ecall 4

        li16 a0, 784       # G5
        li16 a1, 100
        ecall 4

        li16 a0, 987       # B5
        li16 a1, 100
        ecall 4

        li16 a0, 1046      # C6
        li16 a1, 200       # hold the crown note
        ecall 4

        # — Royal Echo (soft repeat of C6)
        li16 a0, 1046
        li16 a1, 150
        ecall 4

        # — Flourish Descent: A5 → G5 → E5 → C5
        li16 a0, 880       # A5
        li16 a1, 120
        ecall 4

        li16 a0, 784       # G5
        li16 a1, 120
        ecall 4

        li16 a0, 659       # E5
        li16 a1, 150
        ecall 4

        li16 a0, 523       # C5
        li16 a1, 200       # resolution on home
        ecall 4

        # — Final Triumphant Trio: G5 → B5 → C6
        li16 a0, 784       # G5
        li16 a1, 80
        ecall 4

        li16 a0, 987       # B5
        li16 a1, 80
        ecall 4

        li16 a0, 1046      # C6
        li16 a1, 250       # grand finale hold
        ecall 4