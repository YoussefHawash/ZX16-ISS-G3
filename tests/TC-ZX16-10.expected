LUI x6, 0x002
ADDI x6, 0x7e
ECALL 5    # SetAudioVolume
LUI x6, 0x004
ADDI x6, 0x0b
LUI x7, 0x001
ADDI x7, 0x64
ECALL 4    # PlayTone
LUI x6, 0x005
ADDI x6, 0x13
LUI x7, 0x001
ADDI x7, 0x64
ECALL 4    # PlayTone
LUI x6, 0x006
ADDI x6, 0x10
LUI x7, 0x001
ADDI x7, 0x64
ECALL 4    # PlayTone
LUI x6, 0x008
ADDI x6, 0x5b
LUI x7, 0x001
ADDI x7, 0x64
ECALL 4    # PlayTone
LUI x6, 0x008
ADDI x6, 0x16
LUI x7, 0x002
ADDI x7, 0x48
ECALL 4    # PlayTone
LUI x6, 0x008
ADDI x6, 0x16
LUI x7, 0x001
ADDI x7, 0x16
ECALL 4    # PlayTone
LUI x6, 0x007
ADDI x6, 0x70
LUI x7, 0x001
ADDI x7, 0x78
ECALL 4    # PlayTone
LUI x6, 0x006
ADDI x6, 0x10
LUI x7, 0x001
ADDI x7, 0x78
ECALL 4    # PlayTone
LUI x6, 0x005
ADDI x6, 0x13
LUI x7, 0x001
ADDI x7, 0x16
ECALL 4    # PlayTone
LUI x6, 0x004
ADDI x6, 0x0b
LUI x7, 0x002
ADDI x7, 0x48
ECALL 4    # PlayTone
LUI x6, 0x006
ADDI x6, 0x10
LUI x7, 0x001
ADDI x7, 0x50
ECALL 4    # PlayTone
LUI x6, 0x008
ADDI x6, 0x5b
LUI x7, 0x001
ADDI x7, 0x50
ECALL 4    # PlayTone
LUI x6, 0x008
ADDI x6, 0x16
LUI x7, 0x002
ADDI x7, 0x7a
ECALL 4    # PlayTone
ADD x0, x0
ADD x0, x0